src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ReadStack.cs

```diff
@@ -37,9 +37,6 @@ internal struct ReadStack
         /// </summary>
         private int _continuationCount;
 
-        // State cache when deserializing objects with parameterized constructors.
-        private List<ArgumentState>? _ctorArgStateCache;
```

Comment:
As a first pass at this, I removed the "caching" altogether. The caching layer wasn't doing anything because we were never re-using the objects.

If we want to do object pooling for `ArgumentState` instances, I'd recommend replacing this list with a `Stack<ArgumentState>` which gets pushed during `ReadStack.Pop()` and can be pulled from in `SetConstructorArgumentState()`.  

One reason I didn't immediately pursue that is that `ArgumentState` has a `List` field which appears to get new'd and nulled out during regular use without pooling (in contrast its array fields get rented and returned from the array pool). Therefore, I wasn't sure whether it was worth pooling the state object we weren't going to pool/reuse all of its fields.

Thoughts?

Reply:
Agreed, removing it altogether is an improvement and I don't think trying to fix the cache is particularly worth the added effort.